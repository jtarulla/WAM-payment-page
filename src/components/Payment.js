import React, { Component } from 'react';
import page from '../assets/page-2.png';
import card from '../assets/card.png';
import cardsNet from '../assets/card_networks.png';
import payButton from '../assets/buttons/btn-pay.png';
import backButton from '../assets/buttons/btn-back.png';

export class Payment extends Component {
	continue = e => {
		e.preventDefault();
		this.props.nextStep();
	};

	back = e => {
		e.preventDefault();
		this.props.prevStep();
	};

	render() {
		return (
			<>
				<div className='page-number'>
					<img src={page} alt='page-number' />
				</div>
				<div className='container'>
					<h2>Pago del producto</h2>
					<p>
						Una vez efectuado el pago del producto, recibirás un email con los
						detalles de la compra.
					</p>
					<div className='card-area'>
						<div className='card'>
							<img src={card} alt='card-networks' />
						</div>
						<h3>Tarjeta de crédito</h3>
						<p>
							Se redirigirá a una plataforma de pago. Se trata de un proceso
							seguro. Puede que la validación del pago tarde 24 horas en
							completarse.
						</p>
						<div className='card-net'>
							<img src={cardsNet} alt='card-networks' />
						</div>
					</div>
					<div className='btn-back' onClick={this.back}>
						<img src={backButton} alt='back-button' />
					</div>
					<div className='btn-continue' onClick={this.continue}>
						<img src={payButton} alt='pay-button' />
					</div>
				</div>
			</>
		);
	}
}

export default Payment;
