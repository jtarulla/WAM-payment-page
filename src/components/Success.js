import React, { Component } from 'react';
import page from '../assets/page-3.png';
import success from '../assets/success.png';
import loading from '../assets/loading.gif';

export class Payment extends Component {
	state = {
		loading: true,
		message: null
	};

	async componentDidMount() {
		const url = 'http://www.mocky.io/v2/5e3d41272d00003f7ed95c09';
		const response = await fetch(url);
		const data = await response.json();
		this.setState({ message: data, loading: false });
	}

	render() {
		const { message } = this.state;

		if (this.state.loading) {
		}

		return (
			<>
				<div className='page-number'>
					<img src={page} alt='page-number' />
				</div>
				<div className='container'>
					{this.state.loading ? (
						<div className='loading'>
							<img src={loading} alt='loading' />
						</div>
					) : (
						<div className='success'>
							<img src={success} alt='success' />
							<h2>{message.title}</h2>
							<p> {message.text} </p>
						</div>
					)}
				</div>
			</>
		);
	}
}

export default Payment;
