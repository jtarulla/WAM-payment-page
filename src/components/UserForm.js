import React, { Component } from 'react';
import nextButton from '../assets/buttons/btn-next.png';
import page from '../assets/page-1.png';

export class UserForm extends Component {
	continue = e => {
		e.preventDefault();
		const { values } = this.props;

		if (values.firstName !== '' && values.lastName !== '') {
			this.props.nextStep();
		}
	};

	render() {
		const { values, handleChange } = this.props;
		return (
			<>
				<div className='page-number'>
					<img src={page} alt='page-number' />
				</div>
				<div className='container'>
					<div className='form-area'>
						<h2>Mis Datos</h2>
						<p>
							Revisa los datos y completa aquellos pendientes antes de continuar
							con el proceso.
						</p>
						<br />
						<form>
							<label>
								Nombre
								<input
									type='text'
									id='firstName'
									name='firstName'
									onChange={handleChange('firstName')}
									defaultValue={values.firstName}
									autoFocus={true}
									required
								/>
							</label>
							<label>
								Apellidos
								<input
									type='text'
									id='lastName'
									name='lastName'
									onChange={handleChange('lastName')}
									defaultValue={values.lastName}
									required
								/>
							</label>
						</form>
					</div>

					<div className='btn-continue' onClick={this.continue}>
						<img src={nextButton} alt='pay-button' />
					</div>
				</div>
			</>
		);
	}
}

export default UserForm;
