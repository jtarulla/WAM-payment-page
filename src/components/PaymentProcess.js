import React, { Component } from 'react';
import UserForm from './UserForm';
import Payment from './Payment';
import Success from './Success';

export class PaymentProcess extends Component {
	constructor(props) {
		super(props);
		this.state = {
			step: 1,
			firstName: '',
			lastName: ''
		};
		this.nextStep = this.nextStep.bind(this);
		this.prevStep = this.prevStep.bind(this);
		this.handleChange = this.handleChange.bind(this);
	}

	// Proceed to next step
	nextStep = () => {
		const { step } = this.state;
		this.setState({
			step: step + 1
		});
	};

	// Go back to prev step
	prevStep = () => {
		const { step } = this.state;
		this.setState({
			step: step - 1
		});
	};

	// Handle fields change
	handleChange = input => e => {
		this.setState({ [input]: e.target.value });
	};

	render() {
		const { step } = this.state;
		const { firstName, lastName } = this.state;
		const values = { firstName, lastName };

		switch (step) {
			case 1:
				return (
					<UserForm
						nextStep={this.nextStep}
						handleChange={this.handleChange}
						values={values}
					/>
				);
			case 2:
				return <Payment nextStep={this.nextStep} prevStep={this.prevStep} />;
			case 3:
				return <Success />;

			default:
				return (
					<UserForm
						nextStep={this.nextStep}
						handleChange={this.handleChange}
						values={values}
					/>
				);
		}
	}
}

export default PaymentProcess;
