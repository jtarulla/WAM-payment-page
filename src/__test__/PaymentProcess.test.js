import { shallow, mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import PaymentProcess from '../components/PaymentProcess';
import UserForm from '../components/UserForm';
import Payment from '../components/Payment';
import Success from '../components/Success';

describe('PaymentProcess Component', () => {
	it('renders PaymentProcess component without errors', () => {
		const wrapper = shallow(<PaymentProcess />);
		expect(toJson(wrapper)).toMatchSnapshot();

		const userForm = wrapper.find(UserForm);
		expect(userForm.exists()).toBe(true);

		wrapper.setState({ step: 2 });
		const payment = wrapper.find(Payment);
		expect(payment.exists()).toBe(true);

		wrapper.setState({ step: 3 });
		const success = wrapper.find(Success);
		expect(success.exists()).toBe(true);
	});

	it('should change state when calling nextStep', () => {
		const wrapper = mount(<PaymentProcess />);
		const button = wrapper.find(UserForm);
		button.invoke('nextStep')();
		expect(wrapper.state().step).toEqual(2);
	});

	it('should change state when calling prevStep', () => {
		const wrapper = mount(<PaymentProcess />);
		wrapper.setState({ step: 2 });
		const button = wrapper.find(Payment);
		button.invoke('prevStep')();
		expect(wrapper.state().step).toEqual(1);
	});

	it('should change state when calling handleChange', () => {
		const wrapper = shallow(<PaymentProcess />);
		wrapper.setState({ firstName: 'John' });
		const input = wrapper.find(UserForm);
		input.invoke('handleChange')({
			target: { value: 'John' }
		});
		expect(wrapper.state().firstName).toEqual('John');
	});
});
