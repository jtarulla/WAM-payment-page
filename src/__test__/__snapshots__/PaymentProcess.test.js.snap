// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentProcess Component renders PaymentProcess component without errors 1`] = `
<UserForm
  handleChange={[Function]}
  nextStep={[Function]}
  values={
    Object {
      "firstName": "",
      "lastName": "",
    }
  }
/>
`;
